### PR TITLE
Added reading of executed file name to gdbserver

### PR DESCRIPTION
**Pull request by SrimantaBarua**

### CODE REVIEW

File: libr/core/rtr.c [hunk 1]
```diff
@@ -1158,6 +1161,33 @@ static int r_core_rtr_gdb_cb(libgdbr_t *g, void *core_ptr, const char *cmd,
 			return r_core_cmd (core, cmd, 0);
 		}
 		break;
+	case 'i':
+		switch (cmd[1]) {
+		case 'f':
+		{
+			ut64 off, len, namelen;
+			if (sscanf (cmd + 2, "%"PFMT64x",%"PFMT64x, &off, &len) != 2) {
+				strncpy (out_buf, "E00", 3);
```
Comment (radare) [inline libr/core/rtr.c]: wat, strncpy doesnt null terminates the string, is this what you want?

File: libr/core/rtr.c [hunk 2]
```diff
@@ -1158,6 +1161,33 @@ static int r_core_rtr_gdb_cb(libgdbr_t *g, void *core_ptr, const char *cmd,
...
+				return 0;
+			}
+			namelen = strlen (core->file->desc->name);
+			if (off >= namelen) {
+				out_buf[0] = 'l';
+				return 0;
+			}
+			snprintf (out_buf, max_len - 1, "l%s",
+				  core->file->desc->name + off);
+			if (len < max_len - 1) {
```
Comment (radare) [inline libr/core/rtr.c]: why is len compared here if its not the return vlaue of snprintf?

File: shlr/gdb/src/gdbserver/core.c [hunk 3]
```diff
@@ -50,6 +51,39 @@ static int _server_handle_qOffsets(libgdbr_t *g, gdbr_server_cmd_cb cmd_cb, void
...
+		if ((pid = (int) strtol (ptr, NULL, 16)) <= 0 || pid != g->pid) {
+			return send_msg (g, "E00");
+		}
+	}
+	if (!(ptr = strchr (ptr, ':'))) {
+		return send_msg (g, "E00");
+	}
+	ptr++;
+	snprintf (cmd, sizeof (cmd) - 1, "if%s", ptr);
+	if (!(buf = calloc (buflen, sizeof (char)))) {
```
Comment (radare) [inline shlr/gdb/src/gdbserver/core.c]: sizeof(char) is a verbose 1, just use 1, we dont have anything against integers

File: shlr/gdb/src/gdbserver/core.c [hunk 4]
```diff
@@ -599,6 +633,13 @@ int gdbr_server_serve(libgdbr_t *g, gdbr_server_cmd_cb cmd_cb, void *core_ptr) {
 			}
 			continue;
 		}
+		if (r_str_startswith (g->data, "qXfer:exec-file:read:")) {
+			if ((ret = _server_handle_exec_file_read (g, cmd_cb,
```
Comment (radare) [inline shlr/gdb/src/gdbserver/core.c]: no need to indent that way, just put that in 1 line

File: libr/core/rtr.c (hunk 1 above)
Comment (SrimantaBarua) [inline libr/core/rtr.c]: The buf is `calloc`ed, so it's already null terminated

**Comment by radare** [comment]
i would strcpy here, there not much benefit in doing that confusing line that may break if someone else changes the calloc for a malloc or so. its better to make the code easy to readn and maintain than do 1 byte optimizations when the whole buffer has been memsetted before

**Comment by SrimantaBarua** [comment]
Sorry about that, just checked man, snprintf returns number of bytes that would have been written if it was not truncated. Stupid of me :p

**Pull request merged**